src/EnergyPlus/OutputReports.cc

```diff
@@ -431,31 +431,31 @@ static void WriteDXFCommon(EnergyPlusData &state, InputOutputFile &of, const std
     print(of, Format_710, "Zone Names");
 
     for (int zones = 1; zones <= state.dataGlobal->NumOfZones; ++zones) {
-        print(of, Format_710, format("Zone={}:{}", zones, normalizeName(state.dataHeatBal->Zone(zones).Name)));
+        print<check_syntax(Format_710)>(of, Format_710, fmt::format("Zone={}:{}", zones, normalizeName(state.dataHeatBal->Zone(zones).Name)));
```

Comment:
What is this fanciness? I'm going to have to learn about this.

Reply:
I'm basically being fancy to avoid our custom Fortran Syntax formatter when the format string doesn't include any Fortran syntax ('R', 'S', 'N', 'Z', 'T'). When we have a `static constexpr std::string_view` of a format, we can check that at compile time with:
```cpp
inline constexpr FormatSyntax check_syntax(const std::string_view format_str)
{
    if (is_fortran_syntax(format_str)) {
        return FormatSyntax::Fortran;
    } else {
        return FormatSyntax::FMT;
    }
}```
The returned enum is used by the various `print` functions in IOFiles.hh to switch at compile time between using normal FMT syntax or our custom Fortran formatter.

It is hard to tell how much time is saved by using the normal FMT syntax vs the custom formatter but there could be a difference for some format strings since the custom formatter sometimes has to call itself.

Reply:
TIL: this!